plugins/lsp/lsp-status.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lsp-status";
+  originalName = "lsp-status.nvim";
+  defaultPackage = pkgs.vimPlugins.lsp-status-nvim;
+  maintainers = [ helpers.maintainers.b3nb5n ];
+  settingsOptions =
+    let
+      indicator_opt = helpers.defaultNullOpts.mkStr "" "string to show as diagnostics. If you don't have Nerd/Awesome Fonts you can replace defaults with ASCII chars.";
+    in
+    {
+      kind_labels =
+        helpers.defaultNullOpts.mkAttrsOf types.str ""
+          "An optional map from LSP symbol kinds to label symbols. Used to decorate the current function name.";
+
+      select_symbol = helpers.defaultNullOpts.mkStr "" "An optional handler of the form function(cursor_pos, document_symbol) that should return true if document_symbol (a DocumentSymbol) should be accepted as the symbol currently containing the cursor.";
+
+      current_function = helpers.defaultNullOpts.mkBool true "true if the current function should be updated and displayed in the default statusline component.";
+
+      show_filename = helpers.defaultNullOpts.mkBool true "true if the current function should be updated and displayed in the default statusline component.";
+
+      indicator_ok = indicator_opt;
+      indicator_errors = indicator_opt;
+      indicator_warnings = indicator_opt;
+      indicator_info = indicator_opt;
+      indicator_hint = indicator_opt;
+
+      indicator_separator = helpers.defaultNullOpts.mkStr " " "a string which goes between each diagnostic group symbol and its count.";
+
+      component_separator = helpers.defaultNullOpts.mkStr " " ''a string which goes between each "chunk" of the statusline component (i.e. different diagnostic groups, messages).'';
+
+      diagnostics = helpers.defaultNullOpts.mkBool true "If false, the default statusline component does not display LSP diagnostics.";
+    };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    extraConfigLuaPre = ''
+      do
+        local lsp_status = require('lsp-status')
+        lsp_status.config(${helpers.toLuaObject cfg.settings})
+        lsp_status.register_progress()
+      end
+    '';
+
+    plugins.lsp = {
+      onAttach = "require('lsp-status').on_attach(client)";
```

Comment:
Is this how the plugin's `on_attach` is supposed to be used?

This is calling lsp-status's `on_attach` from lsp-config's `onAttach`.

Hard to tell from upstream's readme, but I get the impression the plugin handles calling `on_attach`

Reply:
yep, towards the bottom of that readme theres a complete example where they show passing the on attach function directly to the lsp setup call
```lua
lspconfig.rust_analyzer.setup({
  on_attach = lsp_status.on_attach,
  capabilities = lsp_status.capabilities
})
```

Reply:
Thanks for explaining, I glossed over that bit!

Maybe a short comment along the lines of "lsp-status requires us to call `on_attach` and set `capabilities` in lspconfig". Could probably be worded better...

Reply:
for sure, added a comment explaining a bit